src/runtime/threading_backend.cc

```diff
@@ -164,9 +172,13 @@ class ThreadGroup::Impl {
       CPU_SET(id, &cpuset);
     }
 #if defined(__ANDROID__)
-    sched_setaffinity(thread, sizeof(cpu_set_t), &cpuset);
+    if (sched_setaffinity(tid, sizeof(cpu_set_t), &cpuset) == -1) {
```

Comment:
per the man page we should check for != 0 here, correct?

Reply:
The man page also specifies that it returns -1 on failure, but that's for glibc. I looked at the bionic (Android C runtime) sources, and it also returns -1 on failure. Bionic tests seem to use a mix of != 0 and == -1.